nx/lib/nx.ex

```diff
@@ -6658,6 +6658,218 @@ defmodule Nx do
+  | 1            | `max(sum(abs(x), axes: 0))`               | as below                          |
+  | -1           | `min(sum(abs(x), axes: 0))`               | as below                          |
+  | 2            | 2-norm                                    | as below                          |
+  | -2           | smallest singular value (not implemented) | as below                          |
+  | other        | -                                         | power(sum(power(abs(x), p)), 1/p) |
+
+  ## Examples
+
+    ### Vector norms
+
+    iex> Nx.norm(Nx.tensor([3, 4]))
+    #Nx.Tensor<
+      f64
+      5.0
+    >
+
+    iex> Nx.norm(Nx.tensor([3, 4]), ord: 1)
+    #Nx.Tensor<
+      f64
+      7.0
+    >
+
+    iex> Nx.norm(Nx.tensor([3, -4]), ord: :inf)
+    #Nx.Tensor<
+      s64
+      4
+    >
+
+    iex> Nx.norm(Nx.tensor([3, -4]), ord: :neg_inf)
+    #Nx.Tensor<
+      s64
+      3
+    >
+
+    iex> Nx.norm(Nx.tensor([3, -4, 0, 0]), ord: 0)
+    #Nx.Tensor<
+      u64
+      2
+    >
+
+    ### Matrix norms
+
+    iex> Nx.norm(Nx.tensor([[3, -1], [2, -4]]), ord: -1)
+    #Nx.Tensor<
+      s64
+      5
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, -2], [2, -4]]), ord: 1)
+    #Nx.Tensor<
+      s64
+      6
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, -2], [2, -4]]), ord: :neg_inf)
+    #Nx.Tensor<
+      s64
+      5
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, -2], [2, -4]]), ord: :inf)
+    #Nx.Tensor<
+      s64
+      6
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, 0], [0, -4]]), ord: :frobenius)
+    #Nx.Tensor<
+      f64
+      5.0
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, 0], [0, -4]]))
+    #Nx.Tensor<
+      f64
+      5.0
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, 4], [0, -4]]), axes: [1], keep_axes: true)
+    #Nx.Tensor<
+      f64[2][1]
+      [
+        [5.0],
+        [4.0]
+      ]
+    >
+
+    ### Error cases
+
+    iex> Nx.norm(Nx.tensor([3, 4]), ord: :frobenius)
+    ** (RuntimeError) expected a 2-D tensor. Got a 1-D tensor.
+
+    iex> Nx.norm(Nx.tensor([3, 4]), ord: :nuclear)
+    ** (RuntimeError) nuclear norm not implemented yet.
+
+    iex> Nx.norm(Nx.tensor([[0], [1]]), ord: :nuclear)
+    ** (RuntimeError) nuclear norm not implemented yet.
+
+    iex> Nx.norm(Nx.tensor([[0], [1]]), ord: -2)
+    ** (RuntimeError) ord: -2 for 2-D tensor not implemented yet.
+  """
+
+  def norm(tensor, opts \\ []) when is_list(opts) do
+    %{shape: s} = t = tensor!(tensor)
+    assert_keys!(opts, [:ord, :axes, :keep_axes])
+
+    default_axes_opts = [axes: nil, keep_axes: false, ord: nil]
+    opts = Keyword.merge(default_axes_opts, opts)
+
+    rank = rank(t)
+
+    unless rank == 1 or rank == 2,
+      do: raise("expected 1-D or 2-D tensor, got tensor with shape #{inspect(s)}")
+
+    {ord, axes_opts} = Keyword.pop(opts, :ord)
+
+    do_p_norm(t, ord, axes_opts)
+  end
+
+  defp do_p_norm(t, nil, opts), do: do_p_norm(t, 2, opts)
+
+  defp do_p_norm(_t, :nuclear, _opts), do: raise("nuclear norm not implemented yet.")
```

Comment:
If we don't implement it yet, maybe we should simply not support this option. :)